client_request.js

```diff
@@ -27,9 +27,10 @@ var safeParse = require('./lib/util.js').safeParse;
 var validateHostPort = require('./lib/util.js').validateHostPort;
 var uuid = require('node-uuid');
 var util = require('util');
+var middleware = require('./lib/middleware');
 
-/* jshint maxparams: 7 */
-function ClientRequest(client, opts, endpoint, head, body, date, callback) {
+/* jshint maxparams: 8 */
+function ClientRequest(client, opts, endpoint, head, body, date, callback, middlewares) {
```

Comment:
'normal' node behavior is to always have the callback as the last function. If you want to make `middlewares` optional you test if `middlewares` is a function and in that case assign middlewares to the callback and then put a default value in the optional field.